### PR TITLE
interface: app permalink embeds

### PR DESCRIPTION
**Pull request by arthyn**
Adds rich embeds for app links in Landscape for easy sharing

Screen Shot 2021-09-16 at 4 25 17 PM

### CODE REVIEW

File: pkg/interface/src/logic/lib/permalinks.ts [hunk 1]
```diff
@@ -80,6 +81,14 @@ export function referenceToPermalink({ reference }: ReferenceContent): Permalink
       link,
       ...reference.graph
     };
+  } else if ('app' in reference) {
+    const { ship, desk } = reference.app;
+    return {
+      type: 'app',
+      link: `web+urbitgraph://app/${ship}/${desk}`,
```
Comment (liam-fitzgerald) [inline pkg/interface/src/logic/lib/permalinks.ts:88]: New link is actually 
```typescript
`web+urbitgraph://${ship}/${desk}`
```

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
Nit about link structure, plus tests are borked

**Comment by tacryt-socryp** [comment]
Ideally these links could have additional path items that would be passed on to the app in question if it were opened.

File: pkg/landscape/sur/post.hoon [hunk 2]
```diff
@@ -29,6 +29,7 @@
 +$  reference
   $%  [%graph group=resource =uid]
       [%group group=resource]
+      [%app =ship =desk path=@ta]
```
Comment (tacryt-socryp) [inline pkg/landscape/sur/post.hoon:32]: Makes more sense as `[%app =ship =desk =path]` than throwing a `@ta` in there, no? @liam-fitzgerald @arthyn

Comment (liam-fitzgerald) [reply pkg/landscape/sur/post.hoon:32]: Yeah absolutely

Comment (arthyn) [reply pkg/landscape/sur/post.hoon:32]: oh cool, yeah that makes sense, didn't know we had a type specifically for that

**Pull request merged**